src/Binding.lua

```diff
@@ -132,8 +184,10 @@ function Binding.create(initialValue)
 			subscriberCount = 0,
 
 			valueTransform = identity,
-			upstreamBinding = nil,
-			upstreamDisconnect = nil,
+			isJoinedBinding = false,
+			upstreamBindings = {},
+			upstreamConnections = {},
+			upstreamBindingCount = 0,
```

Comment:
We only ever use this value to check if the binding has _any_ upstream bindings; we never use the actual count. Could this be a boolean instead?

Reply:
Now checking if the `upstreamBindings` table is empty or not.